src/test/java/com/puppycrawl/tools/checkstyle/AstTreeStringPrinterTest.java

```diff
@@ -154,4 +155,28 @@ public void testAstTreeSingleLineComments() throws Exception {
                 JavaParser.Options.WITH_COMMENTS);
     }
 
+    @Test
+    public void testTextBlocksEscapesAreOneChar() throws Exception {
+        final String inputFilename = "InputAstTreeStringPrinterTextBlocksEscapesAreOneChar.java";
+        final DetailAST ast = JavaParser.parseFile(
+                new File(getNonCompilablePath(inputFilename)), JavaParser.Options.WITHOUT_COMMENTS);
+
+        final DetailAST objectBlockNode = ast.findFirstToken(TokenTypes.OBJBLOCK);
+        final DetailAST variableDefNode = objectBlockNode.findFirstToken(TokenTypes.VARIABLE_DEF);
+        final DetailAST textBlockContentNode =
+                variableDefNode.findFirstToken(TokenTypes.ASSIGN)
+                        .findFirstToken(TokenTypes.EXPR)
+                        .getFirstChild()
+                        .findFirstToken(TokenTypes.TEXT_BLOCK_CONTENT);
+
+        final String textBlockContent = textBlockContentNode.getText();
+
+        assertThat("Text block content contains \"\\n\" as substring",
+                textBlockContent.contains("\\n"), is(false));
+        assertThat("Text block content line terminator is counted as one character",
+                textBlockContent.length(), is(1));
+        assertThat("Text block content contains only a line terminator",
+                textBlockContent.matches("\n"), is(true));
+    }
```

Comment:
@romani @rnveach is there anything else I can show here?

Reply:
I am satisfied. 
`\n` is escaping of AST printer.